reclass/datatypes/parameters.py

```diff
@@ -115,7 +116,7 @@ def _extend_list(self, cur, new, path):
             ret = [cur]
             offset = 1
 
-        for i in xrange(len(new)):
+        for i in range(len(new)):
```

Comment:
The other problem with `past` is that as far as I can tell, it's not available in all versions of python3

```
[root@worksrv ~]# python --version
Python 3.5.2
[root@worksrv ~]# python -c 'import past'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named 'past'
```

I do have to agree with @Rtzq0 though. In addition to the fact that we should not change any behaviors in order to achieve python3 compatibility, I think it's desirable to ensure that behavior is consistent across runtimes.

The best way to do this would probably be to use `xrange` as the preferred function and fall back to using `range` in cases where it is not available. This could be achieved using the check `'xrange' in dir(__builtins__)`